File: Data Structures & Algorithms/longest-consecutive-sequence/submission-0.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        numsSet = set(nums)

        max_length = 0

        for number in nums:
            # check that a number is the start of a sequence
            if (number - 1) not in numsSet:
                length = 0
                while (number + length) in numsSet:
                    length += 1
                    
                max_length = max(length, max_length)

        return max_length
            

            

File: Data Structures & Algorithms/duplicate-integer/submission-0.java
class Solution {
    public boolean hasDuplicate(int[] nums) {
        HashMap<Integer, Integer> dict = new HashMap<>();
        for(int number: nums){
            if(dict.containsKey(number)==false){
                dict.put(number, 1);
            }
            else{
                return true;
            }
        }
        return false;
    }
}